config/cohere.yaml

```diff
@@ -0,0 +1,75 @@
+# Tokenizer configuration
+# -------------------------------------------------------------------------------------------
+tokenizer:
+  type: CohereHFTokenizer
+  params:
+    model_name: Cohere/Command-nightly
+
+
+# -------------------------------------------------------------------------------------------------------------
+# Chat engine configuration
+# -------------------------------------------------------------------------------------------------------------
+chat_engine:
+  params:
+    system_prompt: *system_prompt
+
+  # -------------------------------------------------------------------------------------------------------------
+  # LLM configuration
+  # -------------------------------------------------------------------------------------------------------------
+  llm: &llm
+    type: CohereLLM
+    params:
+      model_name: command
+      # You can add any additional parameters which are supported by the Cohere Co.Chat API. The values set
+      # here will be used in every Co.Chat API call. For example:
+      # prompt_truncation: "AUTO"
+      # citation_quality: "accurate"
+      # temperature: 0.85
+      # Specifying connectors is contrary to Canopy's purpose of searching the Pinecone knowledge base only,
+      # but technically can still be passed like this:
+      # connectors:
+      #  - "web-search"
+      # Uncomment to suppress errors when unrecognized or unsupported model params are sent to CohereLLM.
+      # ignore_unrecognized_params: true
+
+  # --------------------------------------------------------------------
+  # Configuration for the QueryBuilder subcomponent of the chat engine.
+  # --------------------------------------------------------------------
+  query_builder:
+    type: CohereQueryGenerator
+    params: {}
+    llm:
+      <<: *llm
+
+
+  # -------------------------------------------------------------------------------------------------------------
+  # ContextEngine configuration
+  # -------------------------------------------------------------------------------------------------------------
+  context_engine:
+    # -----------------------------------------------------------------------------------------------------------
+    # KnowledgeBase configuration
+    # -----------------------------------------------------------------------------------------------------------
+    knowledge_base:
+      # --------------------------------------------------------------------------
+      # Configuration for the RecordEncoder subcomponent of the knowledge base.
+      # --------------------------------------------------------------------------
+      record_encoder:
+        type: CohereRecordEncoder
+        params:
+          model_name:                   # The name of the model to use for encoding
+            "embed-english-v3.0"
+          batch_size: 100               # The number of document chunks to encode in each call to the encoding model
```

Comment:
An issue I noticed here is that our embed v3 models here uses 1024 dimension by default. Unless otherwise specified, Canopy creates a Pinecone index with 1536 dimensions, the OAI default. Passing the following specifictations below (similar to `config.py`) helps this issue

```
create_index_params:
  # -------------------------------------------------------------------------------------------
  # Initialization parameters to be passed to create a canopy index. These parameters will
  # be used when running "canopy new".
  # -------------------------------------------------------------------------------------------
  metric: cosine
  dimension: 1024
  spec:
    serverless:
      cloud: aws
      region: us-west-2
```

Reply:
@giannis2two, we're actually just in the process of removing `dimension` from `create_index_params` altogether.  
It was always an optional param, as Canopy tries to infer the dimension from the configured `record_encoder `. Now we simply decided it's not even optional anymore... 😄

Reply:
>  as Canopy anyway tries to infer the dimension from the configured record_encoder

@igiloh-pinecone please disregard. Seems like a local configuration issue when I first created the index through the CLI. Thanks for the details!